packages/web-app-files/tests/unit/views/PrivateLink.spec.js

```diff
@@ -1,16 +1,131 @@
+import { shallowMount } from '@vue/test-utils'
+import { getStore, localVue } from './views.setup'
+import PrivateLink from '@files/src/views/PrivateLink.vue'
+
+localVue.prototype.$client.files = {
+  getPathForFileId: jest.fn(() => Promise.resolve('/lorem.txt'))
+}
+
+const backgroundImgPath = 'assets/loginBackground.jpg'
+const loginLogoPath = 'assets/logo.svg'
+
+const $route = {
+  params: {
+    fileId: '2147491323'
+  },
+  meta: {
+    title: 'Resolving private link'
+  }
+}
+
+const selectors = {
+  backgroundImg: '[data-testid="files-private-link-bg"]',
+  pageTitle: '[data-testid="files-private-link-page-title"]',
+  loginLogo: '.oc-login-logo',
+  loader: '[data-testid="files-private-link-loader"]',
+  ocSpinner: 'oc-spinner-stub',
+  errorMessage: '[data-testid="files-private-link-error-msg"]'
+}
+
 describe('PrivateLink view', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
   describe('when the page has loaded successfully', () => {
-    it.todo('should have the background image set')
-    it.todo('should display the page title')
-    it.todo('should display the logo')
-    it.todo('should resolve the provided file id to a path')
+    let wrapper
+    beforeEach(() => {
+      wrapper = getShallowWrapper()
+    })
+
+    it('should have the background image set', () => {
+      const backgroundImg = wrapper.find(selectors.backgroundImg)
+
+      expect(backgroundImg.exists()).toBeTruthy()
+      expect(backgroundImg.attributes().style).toEqual(
+        `background-image: url(${backgroundImgPath});`
+      )
+    })
+    it('should display the page title', () => {
+      const pageTitle = wrapper.find(selectors.pageTitle)
+
+      expect(pageTitle.exists()).toBeTruthy()
+      expect(pageTitle.text()).toEqual($route.meta.title)
+    })
+    it('should display the logo', () => {
+      const loginLogo = wrapper.find(selectors.loginLogo)
+
+      expect(loginLogo.exists()).toBeTruthy()
+      expect(loginLogo.attributes().src).toEqual(loginLogoPath)
+    })
+    it('should resolve the provided file id to a path', () => {
+      expect(localVue.prototype.$client.files.getPathForFileId).toHaveBeenCalledTimes(1)
+    })
   })
 
-  describe('when the page is still loading', () => {
-    it.todo('should display the loading text with the spinner')
+  describe('when the view is still loading', () => {
+    it('should display the loading text with the spinner', () => {
+      const wrapper = getShallowWrapper(true)
+      const loader = wrapper.find(selectors.loader)
+      const ocSpinner = wrapper.find(selectors.ocSpinner)
+
+      expect(loader.exists()).toBeTruthy()
+      expect(loader.text()).toEqual('Resolving private link…')
```

Comment:
you can match snapshot of loader here rather than asserting the text of the loader